api/include/opentelemetry/trace/span.h

```diff
@@ -82,6 +84,42 @@ class Span
   // virtual void AddEvent(nostd::string_view name, core::SteadyTimestamp
   // timestamp, nostd::span<const std::pair<nostd::string_view name, AttributeValue
   // value>> attributes) noexcept = 0;
+  virtual void AddEvent(nostd::string_view name,
+                        core::SystemTimestamp timestamp,
+                        const KeyValueIterable &attributes) noexcept = 0;
+
+  template <class T, nostd::enable_if_t<detail::is_key_value_iterable<T>::value> * = nullptr>
+  void AddEvent(nostd::string_view name,
+                core::SystemTimestamp timestamp,
+                const T &attributes) noexcept
+  {
+    this->AddEvent(name, timestamp, KeyValueIterableView<T>{attributes});
+  }
+
+  template <class T, nostd::enable_if_t<detail::is_key_value_iterable<T>::value> * = nullptr>
+  void AddEvent(nostd::string_view name, const T &attributes) noexcept
+  {
+    this->AddEvent(name, std::chrono::system_clock::now(), KeyValueIterableView<T>{attributes});
```

Comment:
I would really like to delegate this to the SDK so that it's possible to use alternatives to chrono (e.g. `absl::Now`)

Reply:
What if I left this in as the default implementation, but made the method virtual so that the SDK could override and use a different clock?

Reply:
I moved the timestamping to a virtual function so that the SDK can override with a different clock.

Reply:
Thanks!